modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -996,6 +996,34 @@ CV__DNN_EXPERIMENTAL_NS_BEGIN
                              CV_OUT std::vector<int>& indices,
                              const float eta = 1.f, const int top_k = 0);
 
+     class CV_EXPORTS_W_SIMPLE Model : public Net
```

Comment:
> Model

Please don't introduce GOD entities.
It is better to cover specific cases, like detection and/or classification with better configuration. With name of `DetectionModel` or similar.

Link: https://en.wikipedia.org/wiki/God_object

Reply:
As the advantage, we wanted to hide asynchronous inference mangement into the Model class. So would it be a good idea to keep `dnn::Model` with some method like `predict` which works in the way like `forward` but additionally performs preprocessing and can use VideoCapture as a source of inputs.

Reply:
@l-bat, Please create a separate `DetectionModel` class for object detection networks. It'll be inherited from `Model` and has just `detect` method. For `Model` we'll use something like `predict(const Mat& frame)`